gui/widgets/grid.h

```diff
@@ -0,0 +1,234 @@
+ * of the License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ *
+ */
+
+#ifndef GUI_WIDGETS_GRID_H
+#define GUI_WIDGETS_GRID_H
+
+#include "gui/dialog.h"
+#include "gui/widgets/scrollbar.h"
+#include "common/str.h"
+
+#include "image/bmp.h"
+#include "image/png.h"
+#include "graphics/svg.h"
+
+namespace GUI {
+
+class ScrollBarWidget;
+class GridItemWidget;
+class GridWidget;
+
+const Graphics::ManagedSurface *scaleGfx(const Graphics::ManagedSurface *gfx, int w, int h);
+Graphics::ManagedSurface *loadSurfaceFromFile(const Common::String &name);
+int breakText(const Common::String &str, int fitWidth);
+
+// TODO: Add more platforms
+enum Platform {
+	kPlatformDOS,
+	kPlatformAmiga,
+	kPlatformApple2,
+	kPlatformUnknown = -1
+};
+
+// Cut-pasted commands from Launcher to here to make things work
+// TODO: Find a way to put them back in Launcher.cpp
+enum {
+	kStartCmd = 'STRT',
+	kEditGameCmd = 'EDTG',
+	kLoadGameCmd = 'LOAD',
+	kOpenTrayCmd = 'OPTR',
+};
+
+/* GridItemInfo */
+struct GridItemInfo
+{
+	typedef Common::String String;
+
+	String 		engineid;
+	String 		gameid;
+	String 		language;
+	String 		title;
+	Platform 	platform;
+	String 		thumbPath;
+	int 		entryID;
+
+	GridItemInfo(int id, const String &eid, const String &gid
+		,const String &t, const String &l, const String &p)
+		: entryID(id), gameid(gid), engineid(eid), title(t), language(l) {
+		
+		if (p == "pc")
+			platform = kPlatformDOS;
+		else if (p == "amiga")
+			platform = kPlatformAmiga;
+		else if (p == "apple2")
+			platform = kPlatformApple2;
+		else
+			platform = kPlatformUnknown;
+
+		thumbPath = String::format("%s-%s.png", engineid.c_str(), gameid.c_str());
+	}
+};
+
+/* GridItemTray */
+class GridItemTray: public Dialog, public CommandSender {
+	int			 	_entryID;
+	GridWidget 		*_grid;
+	GuiObject 		*_boss;
+public:
+	typedef Common::String String;
+	typedef Common::Array<Common::String> StringArray;
+
+	typedef Common::U32String U32String;
+	typedef Common::Array<Common::U32String> U32StringArray;
+
+	GridItemTray(GuiObject *boss, int x, int y, int w, int h, int entryID, GridWidget *grid);
+	
+	void handleCommand(CommandSender *sender, uint32 cmd, uint32 data) override;
+	void handleMouseDown(int x, int y, int button, int clickCount) override;
+	void handleMouseWheel(int x, int y, int direction) override;
+	void handleMouseMoved(int x, int y, int button) override;
+};
+
+
+/* GridWidget */
+class GridWidget : public ContainerWidget {
+public:
+	typedef Common::String String;
```

Comment:
You could just add at the beginning of the file `using Common::String;`

Reply:
I replaced the repetitive typedefs with 'using' in 7fea81e.